### PR TITLE
Private Helm Git Repo Repository Support

### LINKED ISSUE #914: SPK should support private helm chart repos at service creation
**As a:**
Developer

**I want:**
SPK to automatically create a `access.yaml` file when  defining a helm chart repository in the `spk service create` command

**So that:**
I will not have to manually see my HLD to Manifest pipeline fail when I reference a private helm chart repository. 

**Describe the solution you'd like:**
Ideally I wouldn't want to have to look into the Fabrikate tool repository to [discover](https://github.com/microsoft/fabrikate/blob/master/docs/auth.md#authentication--personal-access-tokens--accessyaml) that I need to do an additional step.

**Acceptance Criteria:**
I can specify a secure ENV VAR that is linked to a variable group and have the HLD to Manifest pipeline inject this ENV VAR when calling `fab install` and `fab generate` commands. `spk` should scaffold an `access.yaml` file that has a mapping of helm chart URLs to ENV VARs that represent an git repository access token 

Ideally I can specify this at the time of `spk service create` 

**Describe alternatives you've considered:**
Manually updating the HLD and manually testing.

**Does this require updates to documentation?:**
Yes

related to #805

### PR DESCRIPTION
**Pull request by evanlouie**
This PR adds the ability to specify an environment variable
`accessTokenVariable` per service tracked in `bedrock.yaml`. This env variable
will automatically be injected into `access.yaml` as a the holder of a PAT to
use while cloning the value specified in `--helm-config-git` in `hld reconcile`.

closes https://github.com/microsoft/bedrock/issues/914

### CODE REVIEW

**Comment by evanlouie** [comment]
@bnookala Note I simplified the dependency injection a lot in the reconcile code

**Review by andrebriggs — CHANGES_REQUESTED** [review]
@evanlouie We should include some documentation updates since the [issue](https://github.com/microsoft/bedrock/issues/914) says
> Does this require updates to documentation?:
Yes

File: src/commands/service/create.decorator.json [hunk 1]
```diff
@@ -77,6 +77,11 @@
       "arg": "--path-prefix-major-version <path-prefix-major-version>",
       "description": "Version to be used in the path prefix; will be used to configure Traefik2 IngressRoutes. ie. 'v1' will result in a path prefix of '/v1/servicename",
       "defaultValue": ""
+    },
+    {
+      "arg": "--access-token-variable <env-variable>",
```
Comment (mtarng) [inline src/commands/service/create.decorator.json]: Do we want to also combine this with --helm-config-* since it's only used for that configuration?

**Comment by mtarng** [comment]
Specifically do we want to add instructions on how to add a custom variable to our hld -> Manifest pipeline? Currently we don't have a story for adding a variable group for hld -> manifest pipeline, instead we only create pipeline variables at pipeline install time. To create a variable, the users would need to edit the pipeline definition to add the env var name that they're passing in with this new command, and map it to whatever pipeline variable they create.

File: src/commands/service/create.decorator.json (hunk 1 above)
Comment (evanlouie) [reply src/commands/service/create.decorator.json]: done

**Comment by evanlouie** [comment]
@andrebriggs I added a guide

File: guides/auth-private-helm-repos.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+Helm chart repository, we can piggy back off the ability for
+[Fabrikate to consume env variables consuming personal access tokens (PATs)](https://github.com/microsoft/fabrikate/blob/master/docs/auth.md).
+
+## Utilizing The Default PAT (ACCESS_TOKEN_SECRET)
+
+By default, all services created via `spk service create` will be setup to
+consume the environment variable `ACCESS_TOKEN_SECRET` -- The PAT used to clone
+the application repository in the HLD-to-Materialized pipeline. So if the PAT
+used to setup your HLD-to-Materialized pipeline has full read access to the
+Azure DevOps organizations your Helm git repositories are stored, the pipeline
```
Comment (bnookala) [inline guides/auth-private-helm-repos.md]: all set to clone helm charts by default.

File: guides/auth-private-helm-repos.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+is all set by default.
+
+## Making The Pipeline Aware of Custom Environment Variables
+
+In order to tell the pipeline that it needs to utilize a PAT stored in a user
+defined environment variable in the HLD-to-Materialized pipeline, the
+environment variable name -- not the PAT itself -- needs to be tracked in the
```
Comment (bnookala) [inline guides/auth-private-helm-repos.md:24]: How would I get this variable to AzDo? Maybe make a mention of adding the env var to the variable group before hand

**Review by bnookala — CHANGES_REQUESTED** [review]

File: guides/auth-private-helm-repos.md (hunk 3 above)
Comment (evanlouie) [reply guides/auth-private-helm-repos.md:24]: hld->materialized pipeline doesn't have a variable group, only the predefined variables we scaffold it with. Added a note/link to azure docs on how to add more variables.

**Review by bnookala — APPROVED** [review]

**Comment by andrebriggs** [comment]
@evanlouie thanks for adding a walkthrough. We need to update the operational level doc for `spk service create` [here](https://github.com/CatalystCode/spk/blob/master/src/commands/service/create.md)

**Comment by evanlouie** [comment]
@andrebriggs added some top level docs to that page as well as a link to the guide.

**Review by andrebriggs — APPROVED** [review]

**Review by bnookala — APPROVED** [review]

**Pull request merged**